### ISSUE TITLE
systemd doesn't detect webhook as running

### ISSUE THREAD

**Issue by brandonweeks** [acidprime/r10k#207]
After the latest systemd unit file changes, I'm now running into problems with systemd detecting the webhook process as running. As you can see below, systemd doesn't think the process is running even though it is, which causes puppet to try and restart the webhook every run.

[brandon.weeks@foreman ~]$ sudo pkill webhook
[brandon.weeks@foreman ~]$ ps aux | grep webhook
brandon+ 23701  0.0  0.0 112640   964 pts/0    R+   01:25   0:00 grep --color=auto webhook
[brandon.weeks@foreman ~]$ sudo systemctl start webhook.service 
[brandon.weeks@foreman ~]$ ps aux | grep webhook
root     23716  0.0  0.9 221496 18484 ?        Sl   01:26   0:00 /usr/bin/ruby /usr/local/bin/webhook
brandon+ 23729  0.0  0.0 112640   964 pts/0    R+   01:26   0:00 grep --color=auto webhook
[brandon.weeks@foreman ~]$ sudo systemctl status webhook.service -l
webhook.service - R10K Webhook Service
   Loaded: loaded (/usr/lib/systemd/system/webhook.service; enabled)
   Active: inactive (dead) since Sat 2015-06-27 01:26:14 PDT; 24s ago
  Process: 23707 ExecStart=/usr/local/bin/webhook (code=exited, status=0/SUCCESS)
 Main PID: 23707 (code=exited, status=0/SUCCESS)

Jun 27 01:26:08 foreman systemd[1]: Starting R10K Webhook Service...
Jun 27 01:26:08 foreman systemd[1]: Started R10K Webhook Service.
[brandon.weeks@foreman ~]$ 

Here is the same test with the forking unit file:

[brandon.weeks@foreman ~]$ sudo pkill webhook
[brandon.weeks@foreman ~]$ ps aux | grep webhook
brandon+ 25082  0.0  0.0 112640   964 pts/0    R+   01:39   0:00 grep --color=auto webhook
[brandon.weeks@foreman ~]$ sudo systemctl start webhook.service 
[brandon.weeks@foreman ~]$ ps aux | grep webhook
root     25098  0.0  0.9 221436 18628 ?        Sl   01:39   0:00 /usr/bin/ruby /usr/local/bin/webhook
brandon+ 25107  0.0  0.0 112640   964 pts/0    S+   01:39   0:00 grep --color=auto webhook
[brandon.weeks@foreman ~]$ sudo systemctl status webhook.service -l
webhook.service - r10k Webhook Service
   Loaded: loaded (/usr/lib/systemd/system/webhook.service; enabled)
   Active: active (running) since Sat 2015-06-27 01:39:48 PDT; 16s ago
  Process: 25091 ExecStart=/usr/local/bin/webhook (code=exited, status=0/SUCCESS)
 Main PID: 25098 (webhook)
   CGroup: /system.slice/webhook.service
           └─25098 /usr/bin/ruby /usr/local/bin/webhook

Jun 27 01:39:43 foreman systemd[1]: Starting r10k Webhook Service...
Jun 27 01:39:48 foreman systemd[1]: PID 23716 read from file /var/run/webhook/webhook.pid does not exist.
Jun 27 01:39:48 foreman systemd[1]: webhook.service: Supervising process 25098 which is not our child. We'll most likely not notice when it exits.
Jun 27 01:39:48 foreman systemd[1]: Started r10k Webhook Service.
[brandon.weeks@foreman ~]$

**Comment by brandonweeks** [comment]
efe8eb068d actually made RHEL _stop_ working for me. Do you have any more information on the types of environments the forking systemd file didn't work in?

**Comment by joshuabaird** [comment]
I'm seeing the same behavior on RHEL7.

**Comment by brandonweeks** [comment]
This test was on RHEL 7 as well, forgot to specify.

**Comment by joshuabaird** [comment]
This is what seems to work for me:

```
[Unit]
Description=R10K Webhook Service
After=syslog.target network.target

[Service]
Type=forking
User=peadmin
TimeoutStartSec=90
TimeoutStopSec=30
PIDFile=/var/run/webhook/webhook.pid
ExecStart=/usr/local/bin/webhook

KillMode=process

StandardOutput=syslog

[Install]
WantedBy=multi-user.target
```
I'm able to cleanly stop/start the service, and systemd detects it as running (or not) properly.

**Comment by pdaukintis** [comment]
I am having the same issue on Centos 7. Also, systemd unit file above seems to solve this problem as well

**Comment by brandonweeks** [comment]
It would be better to just revert efe8eb0 wholesale if we agree c776fd3 works fine.

**Comment by poikilotherm** [comment]
@brandonweeks
Just my 2 cents: I don't think that's a good idea - @acidprime reverted that for good reason, so maybe change only one thing at a time.

@pdaukintis: could you release a new forge version ASAP please? This is a blocker for my all new puppetmaster... :-/ (And at least this one time I try to not use master branches of modules anymore... ;) )

**Comment by joshuabaird** [comment]
@poikilotherm You can always fork acidprime/r10k or just use pdaukintis/r10k until the fix is merged and released here.

**Comment by poikilotherm** [comment]
Thanks for your suggestion. But depending on forks is not any better than depending on master branches... :-D

**Comment by acidprime** [comment]
#218 please re-open if still having issues

**Comment by acidprime** [comment]
released in v2.8.3

**Comment by acidprime** [comment]
Also sorry for the flip flopping here folks I really need beaker coverage for this, I going to keep the split but 2.8.3 changes to forking as that was tested on RHEL 7 for this issue.
